### ISSUE TITLE
Images with spaces in filename don't work

### ISSUE THREAD

**Issue by TimonVS** [terrymun/Fluidbox#112]
Images with spaces in the filename don't work because the background url's are set without double quotes.

**Comment by terrymun** [comment]
Please see #58.

Basically, spaces in an image path name is *invalid* and it should be manually escaped (using the `%20` HTML entity) before entering it in the markup. This is the same as using `&` instead of `&amp;` — the browser will automatically fix this to make things work, but that does not mean that it is the right way to do it.

**Comment by TimonVS** [comment]
@terrymun I can't help it that the CMS I use (Statamic) doesn't prettify the image urls, and their opinion on this is that it's the task of a browser to do this (https://lodge.statamic.com/public-house/1261-transform-tag-friendly-image-url), but the browsers I tested (Chrome and Safari) don't do this.

I hope you'll reconsider merging in the fix.

**Comment by terrymun** [comment]
Unfortunately, that means the developers over at Statamic needs to take a basic lesson in HTML standards and read the [RFC 2396 standard](http://tools.ietf.org/html/rfc2396). I suggest you take this issue up with the devs of the CMS—and if they don't want to abide with vetted web standards, get another one that does so. No CMS should think they're above the standards.

Whitespaces are specifically slated to be escaped, because multiple whitespaces will be collapsed in a HTML document, and a single whitespace in a file name can mean one, or more, whitespace characters in the original file name.

> The space character is excluded because significant spaces may disappear and insignificant spaces may be introduced when URI are transcribed or typeset or subjected to the treatment of word-processing programs.  Whitespace is also used to delimit URI in many contexts.

Some browsers will not even attempt to fix the whitespace in image URLs present in HTML (Firefox, Chrome, Safari will not do so), because it can be of ambiguous length. Asking Fluidbox to emulate the invalid behaviour of assuming that a single whitespace is as intended is skirting the established URI standards.

However, if you really need Fluidbox implemented with Statamic, simply use `encodeURI()` before passing the image URI into the `background-image` attribute, or use double quotes to wrap your plugins. The problem is that using double quotes is that double-quoted characters in the URI will still require to be escaped, albeit less of a common occurrence, will still have the chance of appearing.

In other words, wrapping in quotes is *not an all encompassing panacea* for invalid HTTP requests made by Fluidbox. The issue lies with how the markup is generated, which should contain properly escaped URIs by RFC 2398, anyway.

**Comment by TimonVS** [comment]
@terrymun Okay, your explanation makes sense now. I guess I'll just add a filter to Statamic to fix the urls for now.
